src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -595,7 +596,17 @@ fn env_args(
         return Ok(Vec::new());
     }
 
-    // First try RUSTFLAGS from the environment
+    // First try CARG_ENCODED_RUSTFLAGS from the environment.
+    // Prefer this over RUSTFLAGS since it's less prone to encoding errors.
+    if let Ok(a) = env::var(format!("CARGO_ENCODED_{}", name)) {
+        let args = a
+            .split('\x1f')
+            .filter(|s| !s.is_empty())
```

Comment:
This seems surprising, and technically an empty argument should be a valid possibility. Did this arise because of the ambiguity with having no arguments at all? If so, maybe we should use it like a terminator, so `"\x1f"` is a single empty arg while `""` means no arguments.

Reply:
Or just treat an entire empty input string as no arguments, and split normally otherwise.

That's not to say I have any real use for an empty argument, but if we're trying to encode nicely, that filter feels like a kludge.

Reply:
Yup, `split` returns a single item of an empty string when given an empty string as input. Will fix that the way you suggest instead 👍